app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt

```diff
@@ -154,9 +161,118 @@ class LocalSource(
                 }
             }
 
+        // get Metadata from ComicInfo.xml Files
+        val xmlFile = getMangaDirsFiles(manga.url, baseDirsFile)
+            .firstOrNull { it.extension.equals("xml", ignoreCase = true) }
+
+        val noXmlFile = getMangaDirsFiles(manga.url, baseDirsFile)
+            .firstOrNull { it.extension.equals("noxml", ignoreCase = true) }
+
+        try {
+            val encoding = "UTF-8"
+            if (xmlFile != null) {
+                val comicInfo = xml.decodeFromReader<ComicInfo>(AndroidXmlReader(xmlFile.inputStream(), encoding))
+                if (comicInfo.title != null || comicInfo.writer != null || comicInfo.summary != null ||
+                    comicInfo.genre != null || comicInfo.penciller != null
+                ) setMetadataXml(comicInfo, manga)
+            } else if (noXmlFile == null) {
+                var createNoXmlFile = true
+                val mangaDir = getMangaDir(manga.url, baseDirsFile)
+                val folderPath = mangaDir?.absolutePath
+                val fileName = "ComicInfo.xml"
+
+                val chapterArchives = getMangaDirsFiles(manga.url, baseDirsFile)
+                    .filter { isSupportedFile(it.extension) && !it.extension.equals("epub", true) }
+                    .toList()
+
+                for (chapter in chapterArchives) {
+                    if (chapter.extension.equals("zip", true) || chapter.extension.equals("cbz", true)) {
+                        val zip = withContext(Dispatchers.IO) { ZipFile(chapter) }
+                        val zipEntry = zip.getEntry("ComicInfo.xml")
+
+                        if (zipEntry != null) {
+                            val archiveInputStream = withContext(Dispatchers.IO) { zip.getInputStream(zipEntry) }
+                            val comicInfo = xml.decodeFromReader<ComicInfo>(AndroidXmlReader(archiveInputStream, encoding))
+                            if (comicInfo.title != null || comicInfo.writer != null || comicInfo.summary != null ||
+                                comicInfo.genre != null || comicInfo.penciller != null
+                            ) setMetadataXml(comicInfo, manga)
+                            withContext(Dispatchers.IO) { archiveInputStream.close() }
```

Comment:
Instead of closing the stream manually use the function called `use` and it will handle it for you

```suggestion
                                withIOContext {
                                    zip.getInputStream(zipEntry).use { archiveInputStream ->
                                        val comicInfo = xml.decodeFromReader<ComicInfo>(AndroidXmlReader(archiveInputStream, encoding))
                                        if (comicInfo.title != null || comicInfo.writer != null || comicInfo.summary != null ||
                                            comicInfo.genre != null || comicInfo.penciller != null) {
                                            setMetadataXml(comicInfo, manga)
                                        }
                                    }
                                }
```